packages/harpguru-core/src/components/harp-cell/harp-cell.tsx

```diff
@@ -50,24 +51,35 @@ export const HarpCell = ({ yxCoord }: HarpCellProps): React.ReactElement => {
   const handleTapStateChange = ({
     nativeEvent,
   }: TapGestureHandlerStateChangeEvent) => {
-    if (nativeEvent.state !== State.END) return
-
-    addBufferedActivityToggle(thisDegreeId)
+    const interestingTouchStates = [State.BEGAN, State.CANCELLED, State.FAILED]
+    if (interestingTouchStates.includes(nativeEvent.state)) {
+      setIsTouchState(nativeEvent.state)
+    }
+    if (nativeEvent.state === State.END) {
+      addBufferedActivityToggle(thisDegreeId)
+    }
   }
 
+  React.useEffect(() => {
+    if (touchState !== State.UNDETERMINED) {
+      setIsTouchState(State.UNDETERMINED)
+    }
+  }, [thisIsActiveId, setIsTouchState])
+
   const harpCellAccessibleProps = {
     degreeId: thisDegreeId,
     pitchId: thisPitchId,
     isActive: thisIsActiveId === IsActiveIds.Active,
     displayMode: activeDisplayMode,
     activeExperienceMode: activeExperienceMode,
     baseStyles: baseHarpCellStyles,
+    touchState,
```

Comment:
Sort out the ordering of this object. It should match the type definition in harp-cell-accessible

Reply:
Done